src/entries/popup/pages/home/Points/ClaimOverview.tsx

```diff
@@ -0,0 +1,403 @@
+    sellTokenAddress: opEth?.address as Address,
+    buyTokenAddress: destinationEth?.address as Address,
+    sellAmount: claimableBalance.amount || 0,
+    buyAmount: '',
+  });
+
+  const { mutate: claimRewards, isSuccess: claimSuccess } = useMutation<
+    ClaimUserRewardsMutation['claimUserRewards']
+  >({
+    mutationFn: async () => {
+      const response =
+        process.env.IS_TESTING === 'true'
+          ? CLAIM_MOCK_DATA
+          : await metadataPostClient.claimUserRewards({ address });
+      const claimInfo = response?.claimUserRewards;
+
+      if (claimInfo?.error) {
+        setError(claimInfo?.error.type);
+      }
+
+      // clear and refresh claim data so available claim UI disappears
+      invalidatePointsQuery(address);
+      refetchPointsQuery(address);
+
+      return claimInfo;
+    },
+    onSuccess: async (d) => {
+      if (
+        claimNetwork !== ChainId.optimism &&
+        d?.txHash &&
+        opEth &&
+        destinationEth
+      ) {
+        const nonce = await getNextNonce({
+          address,
+          chainId: ChainId.optimism,
+        });
+        const quote = await getCrosschainQuote({
+          chainId: ChainId.optimism,
+          fromAddress: address,
+          sellTokenAddress: opEth.address,
+          buyTokenAddress: destinationEth.address,
+          sellAmount: claimAmount || '0',
+          slippage: swapSlippage?.slippagePercent || 2,
+          destReceiver: address,
+          swapType: SwapType.crossChain,
+          toChainId: claimNetwork,
+          feePercentageBasisPoints: 0,
+        });
+        const actionParams: RapActionParameters = {
+          sellAmount: claimableBalance.amount,
+          chainId: ChainId.optimism,
+          assetToSell: opEth,
+          assetToBuy: destinationEth,
+          quote: quote as QuoteTypeMap['crosschainSwap'],
+          flashbots: false,
+          claimHash: d?.txHash,
+          nonce,
+        };
+        const { errorMessage, nonce: bridgeNonce } = await wallet.executeRap({
+          rapActionParameters: actionParams,
+          type: 'crosschainSwap',
+        });
+
+        if (errorMessage) {
+          setBridgeError(errorMessage);
+        }
+
+        if (typeof bridgeNonce === 'number') {
+          setBridgeSuccess(true);
+        }
+      }
+    },
+  });
+
+  const backToHome = () => {
+    navigate(ROUTES.HOME, {
+      state: { tab: 'points', skipTransitionOnRoute: ROUTES.HOME },
+    });
+  };
+
+  const claimNeedsBridge = claimNetwork !== ChainId.optimism;
+  const claimFinished = claimNeedsBridge ? bridgeSuccess : claimSuccess;
+  const showPreparingClaim = !showSummary;
+  const showSuccess =
+    !waitToDisplay && claimFinished && !showSummary && !error && !bridgeError;
+  const showError = !waitToDisplay && (error || bridgeError);
+
+  useEffect(() => {
+    claimRewards();
+    setTimeout(() => setWaitToDisplay(false), 3000);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
will this execute twice if you reload the extension? considering this is a routed path         path: ROUTES.CLAIM_OVERVIEW,

Reply:
refactored, claimRewards is now called from the parent on network selection